ROADMAP.md

```diff
@@ -0,0 +1,24 @@
+# etcd roadmap
+
+This document defines high level goals for project.
+
+## Milestones
```

Comment:
We should be careful about redefining meaning of the words. 

So far in etcd we were using milestones = future minor release:   v3.5, v3.6.
Here we naming as milestone a focus area we want to invest. 

Let's use consistent terms.  My proposal it to think about this as 3 level hierarchy:  

  1. Milestones -> are milestones as defined in https://github.com/etcd-io/etcd/milestones. Let's keep them as publicly visible releases (might be patch). 
  
  2. Efforts
       - are bugs that track progress on multiple issues that need to be addressed with a common objective
       - [or] (for bigger efforts) projects: https://github.com/etcd-io/etcd/projects. 
         Still I would represent project as umbrella issue -> as it seems that project cannot be assigned to a milestone. 
   3. Issues - for individual work items.
   
 Now the question remains: 
 If we have a tool to dynamically track the milestones with attached efforts / items, do we need to redundantly track it in a markdown doc ? 
 
And I would say - we don't.  I assume that the purpose of the doc is different. It's a statement of intent what we want to focus in following releases. And thanks to being submitted by maintainers and reviewed, it forces them to be on the same page (as opposed to an individual maintainer assigning an issue to a milestone). But If that's the goal, let's call it explicitly in the preamble to this doc. 

Then let's have: 

## Milestones:
 
### [release-v3.6](https://github.com/etcd-io/etcd/milestone/38)

The main focus of the v3.6 is the reduction of technical debts. The explicit goal is to avoid new features. 
The focus will be on: 
  1.  deprecate/decomission experimental / legacy cody:
       - decommission storage v2 (link to a tracking bug)
       - experimental features are graduated or removed  (link to a tracking bug)
  3. ...

### [release-v3.5.x](https://github.com/etcd-io/etcd/milestone/38)

The same as release-v3.4.x.

### [release-v3.4.x](https://github.com/etcd-io/etcd/milestone/38)

The release focuses on stability. Etcd maintainers are going to backport: 
   - critical/important vulnerabilities
   - backportable test coverage (including robustness tests)
   - critical correctness or robustness issue fixes
   - not invasive flags/warnings that enable easier transition to next release.